File: demo/lang/python/3_http_inteface_call.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
'''
[case]

title=check remote interface response
cid=0
pid=0

[group]
1. Send a request to interface http://xxx
2. Retrieve sessionID field from response json
3. Check its format >> `^[a-z0-9]{26}`

[esac]
'''

import requests
import json

jsonStr = requests.get('http://pms.zentao.net/?mode=getconfig').content #need requests library (pip/pip3 install requests)
jsonObj = json.loads(jsonStr)
print('>> ' + jsonObj['sessionID'])

File: demo/lang/javascript/3_http_inteface_call.js
#!/usr/bin/env node
/**
title=check remote interface response
cid=0
pid=0

1. Send a request to interface http://xxx
2. Retrieve sessionID field from response json
3. Validate its format >> `^[a-z0-9]{8}`

*/

const https = require('https');

https.get('https://cn.bing.com/HPImageArchive.aspx?format=js&idx=0&n=1', function(req) {
    let jsonStr = '';

    req.on('data', function(data) {
        jsonStr += data;
    });
    req.on('end', () => {
        if(req.statusCode === 200){
            try{
                const json = JSON.parse(jsonStr);
                console.log(json.images[0].startdate)
            } catch(err){
                console.log('ERR: ' + err);
            }
        }
    });
});
